File: examples/tcp-probe.dart
import 'dart:io';
import 'dart:async';

// OpenClaw TCP port probe (Dart) — checks gateway nodes
Future<bool> probe(String host, int port,
    {Duration timeout = const Duration(seconds: 3)}) async {
  try {
    final socket = await Socket.connect(host, port, timeout: timeout);
    socket.destroy();
    return true;
  } catch (_) {
    return false;
  }
}

Future<void> main() async {
  final nodes = [
    ['localhost', 8080],
    ['localhost', 8081],
  ];
  for (final n in nodes) {
    final ok = await probe(n[0] as String, n[1] as int);
    print('${ok ? "OK  " : "FAIL"} ${n[0]}:${n[1]}');
  }
}

File: examples/tcp-probe.ts
// OpenClaw TCP port probe (TypeScript / Node) — checks gateway nodes
import * as net from "node:net";

function probe(host: string, port: number, timeout = 3000): Promise<boolean> {
  return new Promise((resolve) => {
    const socket = new net.Socket();
    const done = (ok: boolean) => {
      socket.destroy();
      resolve(ok);
    };
    socket.setTimeout(timeout);
    socket.once("connect", () => done(true));
    socket.once("timeout", () => done(false));
    socket.once("error", () => done(false));
    socket.connect(port, host);
  });
}

const nodes: [string, number][] = [["localhost", 8080], ["localhost", 8081]];

(async () => {
  for (const [host, port] of nodes) {
    const ok = await probe(host, port);
    console.log(`${ok ? "OK  " : "FAIL"} ${host}:${port}`);
  }
})();
